napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1620 @@
+from ._constants import (Mode, BOX_LINE_HANDLE, BOX_LINE, BOX_TOP, BOX_CENTER,
+                         BOX_LEN, BOX_HANDLE, BOX_WITH_HANDLE)
+from .shape_list import ShapeList
+from .shape_util import create_box, point_to_lines
+from .shapes import Rectangle, Ellipse, Line, Path, Polygon
+
+
+@add_to_viewer
+class Shapes(Layer):
+    """Shapes layer.
+
+    Parameters
+    ----------
+    data : np.array | list
+        List of np.array of data or np.array. Each element of the list
+        (or row of a 3D np.array) corresponds to one shape. If a 2D array is
+        passed it corresponds to just a single shape.
+    shape_type : string | list
+        String of shape shape_type, must be one of "{'line', 'rectangle',
+        'ellipse', 'path', 'polygon'}". If a list is supplied it must be the
+        same length as the length of `data` and each element will be applied to
+        each shape otherwise the same value will be used for all shapes.
+    edge_width : float | list
+        thickness of lines and edges. If a list is supplied it must be the same
+        length as the length of `data` and each element will be applied to each
+        shape otherwise the same value will be used for all shapes.
+    edge_color : str | tuple | list
+        If string can be any color name recognized by vispy or hex value if
+        starting with `#`. If array-like must be 1-dimensional array with 3 or
+        4 elements. If a list is supplied it must be the same length as
+        the length of `data` and each element will be applied to each shape
+        otherwise the same value will be used for all shapes.
+    face_color : str | tuple | list
+        If string can be any color name recognized by vispy or hex value if
+        starting with `#`. If array-like must be 1-dimensional array with 3 or
+        4 elements. If a list is supplied it must be the same length as
+        the length of `data` and each element will be applied to each shape
+        otherwise the same value will be used for all shapes.
+    opacity : float | list
+        Opacity of the shapes, must be between 0 and 1.
+    z_index : int | list
+        Specifier of z order priority. Shapes with higher z order are displayed
+        ontop of others. If a list is supplied it must be the same length as
+        the length of `data` and each element will be applied to each shape
+        otherwise the same value will be used for all shapes.
+    name : str, keyword-only
+        Name of the layer.
+
+    Attributes
+    ----------
+    data : ShapeList
+        Object containing all the shape data.
+    edge_width : float
+        thickness of lines and edges.
+    edge_color : str
+        Color of the shape edge.
+    face_color : str
+        Color of the shape face.
+    opacity : float
+        Opacity value between 0.0 and 1.0.
+    selected_shapes : list
+        List of currently selected shapes.
+    mode : Mode
+        Interactive mode.
+    _mode_history : Mode
```

Comment:
Can you move all the private attributes to an Extended Summary section? Search for "private" in https://docs.scipy.org/doc/numpy-1.15.0/docs/howto_document.html . Currently this seems extremely onerous where it doesn't have to be.

Reply:
I have changed this in the model - I can change it in other files too if you want

Reply:
@sofroniewn yes i think would be good to have consistency on this

Reply:
done